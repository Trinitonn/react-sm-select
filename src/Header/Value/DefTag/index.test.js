import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { DefTag } from './index';

const methods = {
  onTagRemove: jest.fn(),
};

describe('DefTag', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <DefTag
        label="One"
        index={2}
        removableTag={true}
        {...methods}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly without close', () => {
    const wrapper = shallow(
      <DefTag
        label="One"
        index={2}
        {...methods}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call onTagRemove on close click', () => {
    const wrapper = shallow(
      <DefTag
        label="One"
        index={2}
        removableTag={true}
        {...methods}
      />
    );
    wrapper.find('.Header__tag__close').prop('onClick')('event');
    expect(methods.onTagRemove).toHaveBeenCalledWith(2, 'event');
  });
});
