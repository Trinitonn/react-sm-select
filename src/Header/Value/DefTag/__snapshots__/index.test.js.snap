// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefTag should render properly 1`] = `
<div
  className="Header__tag"
>
  <div
    className="Header__tag__label"
  >
    One
  </div>
  <div
    className="Header__tag__close"
    onClick={[Function]}
  >
    âœ•
  </div>
</div>
`;

exports[`DefTag should render properly without close 1`] = `
<div
  className="Header__tag"
>
  <div
    className="Header__tag__label"
  >
    One
  </div>
</div>
`;
